fdbbackup/backup.actor.cpp

```diff
@@ -3297,6 +3297,130 @@ Optional<Database> connectToCluster(std::string const& clusterFile,
+	                                                              g_rgBackupExpireOptions,
+	                                                              g_rgBackupDeleteOptions,
+	                                                              g_rgBackupDescribeOptions,
+	                                                              g_rgBackupDumpOptions,
+	                                                              g_rgBackupTagsOptions,
+	                                                              g_rgBackupListOptions,
+	                                                              g_rgBackupQueryOptions,
+	                                                              g_rgRestoreOptions,
+	                                                              g_rgDBAgentOptions,
+	                                                              g_rgDBStartOptions,
+	                                                              g_rgDBStatusOptions,
+	                                                              g_rgDBSwitchOptions,
+	                                                              g_rgDBAbortOptions,
+	                                                              g_rgDBPauseOptions };
+
+// The last element in SOption arrays is always END_MARKER = SO_END_OF_OPTIONS.
+constexpr CSimpleOpt::SOption END_MARKER = SO_END_OF_OPTIONS;
+
+
+/**
+ * Validates and processes a command-line option.
+ *
+ * This function checks if the current argument (argv[i]) matches any known option.
+ * If the option requires a parameter, it consumes the next argument as the parameter.
+ * The processed option (and its parameter, if any) are added to the 'options' vector.
+ *
+ * Parameters:
+ *   argc    - Total number of command-line arguments.
+ *   argv    - Array of command-line argument strings.
+ *   i       - Current index in argv; incremented if a parameter is consumed.
+ *   options - Vector to which valid options (and their parameters, if any) are appended.
+ *
+ * Returns:
+ *   true if the option is recognized and valid (and its parameter, if required, is present);
+ *   false otherwise.
+ *
+ * This function is used to reorder and validate command-line arguments.
+ */
+static bool processOption(int argc, char* argv[], int& i, std::vector<char*>& options) {
+	std::string_view option = argv[i];
+
+	options.emplace_back(argv[i]);
+	size_t equalPos = option.find('=');
+
+	if (equalPos != std::string_view::npos) {
+		option = option.substr(0, equalPos);
+	}
+
+	for (auto* opt : allOptionArrays) {
+		for (int j = 0; opt[j].nId != END_MARKER.nId; ++j) {
+			const char* knownOpt = opt[j].pszArg;
+			size_t knownOptLen = strlen(knownOpt);
+			bool isPrefixOpt = knownOptLen > 1 && knownOpt[knownOptLen - 1] == '-';
+
+			if (option == knownOpt || (isPrefixOpt &&
+									   option.size() >= knownOptLen &&
+									   option.compare(0, knownOptLen, knownOpt) == 0))
+			{
+				if (opt[j].nArgType == SO_REQ_SEP && equalPos == std::string_view::npos) {
```

Comment:
We do SO_REQ_SEP only in here. Should we be able to handle other types like SO_REQ_JOINED, etc.?